group/group.tf

```diff
@@ -21,3 +21,19 @@ resource "azuread_group_member" "members" {
   group_object_id  = azuread_group.group.object_id
   member_object_id = each.key
 }
+
+resource "azurerm_role_assignment" "roles" {
+  for_each                               = var.role_assignments
+  name                                   = each.value.name
+  scope                                  = each.value.scope == null ? "/subscriptions/${data.azurerm_client_config.current.subscription_id}" : each.value.scope
+  role_definition_id                     = each.value.role_definition_id
+  role_definition_name                   = each.value.role_definition_name
+  principal_id                           = azuread_group.group.object_id
+  condition                              = each.value.condition
+  condition_version                      = each.value.condition_version
+  delegated_managed_identity_resource_id = each.value.delegated_managed_identity_resource_id
+  description                            = each.value.description
+  skip_service_principal_aad_check       = each.value.skip_service_principal_aad_check == null ? false : each.value.skip_service_principal_aad_check
+}
+
+data "azurerm_client_config" "current" {}
```

Comment:
Including `data "azurerm_client_config" "current" {}` in here is a bad idea and negates the previous PRs which aims to reduce the amount of data sources managed with Terraform Cloud.

I suggest this is resolved on caller level (`/groups.tf`)